network/network_read.c

```diff
@@ -69,7 +69,7 @@ callback_buf(void * cookie)
 		goto tryagain;
 
 	/* Invoke the callback and return. */
-	return (docallback(C, C->bufpos));
+	return (docallback(C, (ssize_t)C->bufpos));
```

Comment:
Maybe also add an assertion that bufpos <= SSIZE_MAX?  We have an assertion in network_read but not one close to here.